### ISSUE TITLE
Proxy variant with IDisposable?

### ISSUE THREAD

**Issue by StefanBertels** [louthy/echo-process#3]
Hi Paul,

I'm just checking some options for better type safety (see next issue...) and I'm trying the proxy variant because the non-OO style is not that type-safe for multi-message actors.

The following code works without Dispose/IDisposable, but raises a runtime exception in `spawn`otherwise:

System.TypeLoadException: 'Method 'Dispose' in type 'ProcessTest.IMyStateProxy' from assembly 'ProcessProxies, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.'

```
public interface IMyState : IDisposable
    {
        void Add(int num);
        void Remove(int num);
        void Show();

    }
    class Program
    {
        static void Main(string[] args)
        {
            ProcessConfig.initialise();

            ProxyTest();

            Console.ReadKey();

            shutdownAll();

            Console.WriteLine("end, press key.");
            Console.ReadKey();

        }


        public class MyState: IMyState
        {
            HashSet<int> _state = HashSet<int>();

            public MyState()
            {
                Console.WriteLine("Setup called");
            }

            public void Add(int num)
            {
                Console.WriteLine($"add {num}...");
                _state = _state.Add(num);
            }

            public void Dispose()
            {
                Console.WriteLine($"Dispose");
            }

            public void Remove(int num)
            {
                Console.WriteLine($"Remove {num}...");
                _state = _state.Remove(num);
            }

            public void Show()
            {
                Console.WriteLine("content: " + string.Join(",", _state));
            }

         
        }

        private static void ProxyTest()
        {
            IMyState state = spawn<IMyState>("mystate", () => new MyState());

            state.Add(5);
            state.Add(3);
            state.Show();
            state.Remove(5);
            state.Show();
        }
```

**Comment by louthy** [comment]
Hmm, I thought I'd fixed that.  The issue is that the proxy builder isn't implementing the inherited interface methods, only the methods from `IMyState` (so not `IDisposable.Dispose`).  I'll investigate.

**Comment by louthy** [comment]
Fixed in `v2.0.8`
